src/types.rs

```diff
@@ -84,12 +84,18 @@ impl TMat<Scalar> {
 }
 
 /// 2 dimensional vector
-#[derive(Clone, Copy, Default)]
-pub struct DVec<T>(pub [T; 2]);
+#[derive(Clone, Copy, Default, PartialEq, Eq)]
+pub struct DVec<T>([T; 2]);
```

Comment:
I removed `Copy`. This was quite a good idea because it forced me to have a loser look on how I was managing type arithmetic. I may have saved some copies.